core/templates/domain/exploration/read-only-exploration-backend-api.service.ts

```diff
@@ -61,21 +62,25 @@ export class ReadOnlyExplorationBackendApiService {
 
   constructor(
     private http: HttpClient,
-    private urlInterpolationService: UrlInterpolationService) {}
+    private urlInterpolationService: UrlInterpolationService,
+    private urlService: UrlService) {}
 
   private _fetchExploration(
       explorationId: string, version: number | null
   ): Promise<FetchExplorationBackendResponse> {
     return new Promise((resolve, reject) => {
-      const explorationDataUrl = this._getExplorationUrl(
-        explorationId, version);
-
-      this.http.get<FetchExplorationBackendResponse>(
-        explorationDataUrl).toPromise().then(response => {
-        resolve(response);
-      }, errorResponse => {
-        reject(errorResponse.error.error);
-      });
+      if (!('skill_editor' === this.urlService.getPathname()
```

Comment:
Why do we need this `if` block? I thought the changes in the context service should be sufficient.

Reply:
Without the if block, the `skillId` will be interpolated with the exploration path and console will throw an Error 404 `explorehandler/init/<skillId>` page not found.

Reply:
hm in that case, it is not required to make this call right? Let's add a check before this method is actually called (its better to do checks at the top level if possible, this avoid unnecessary comuptation).